index.js

```diff
@@ -1,43 +1,50 @@
-const { ApolloClient, createBatchingNetworkInterface } = require('apollo-client');
-const gql = require('graphql-tag');
+const fetch = require('isomorphic-fetch');
 const _ = require('lodash');
 
-// Pollyfill: https://github.com/apollographql/apollo-client/issues/1225
-require('isomorphic-fetch'); // eslint-disable-line global-require
-
 let client;
 
-const mergeHeaderArguments = (options, headers) =>
-  _.merge(options, { variables: { __headers: headers } });
+const createQuery = (query, variables) => {
+  const body = {
+    query,
+    variables,
+    operationName: null,
+  };
 
-module.exports.register = (opts, dependencies, next) => { // eslint-disable-line consistent-return
-  if (opts.batchInterval && !_.isInteger(opts.batchInterval)) {
-    return next(new Error('The batchInterval parameter is invalid'));
+  return JSON.stringify(body);
+};
+
+const checkResponseStatus = (response) => {
+  if (response.status >= 200 && response.status < 300) {
+    return response;
   }
 
+  const error = new Error(response.statusText);
+  error.response = response;
+  throw error;
+};
+
+const parseResponse = response => response.json();
+
+module.exports.register = (opts, dependencies, next) => { // eslint-disable-line consistent-return
   if (!opts.serverUrl) {
     return next(new Error('The serverUrl parameter is invalid'));
   }
 
-  const options = { uri: opts.serverUrl, batchInterval: opts.batchInterval || 10 };
-  const networkInterface = createBatchingNetworkInterface(options);
-
-  networkInterface.use([{
-    applyMiddleware(req, done) {
-      if (!req.options.headers) {
-        // eslint-disable-next-line
-        req.options.headers = req.request.variables.__headers;
-      }
-      done();
-    },
-  }]);
+  client = (options, headers) => fetch(opts.serverUrl, {
```

Comment:
I think that as OC registry owner, I great feature would be to possibly set a timeout as plugin option, so that I am sure we handle slow requests on the plugin level. What do you think?